File: src/main/kotlin/com/lcaohoanq/sp/domains/notifications/FcmTokenPort.java
package com.lcaohoanq.sp.domains.notifications;

public interface FcmTokenPort {

    record UserDeviceTokenDto(
        Long id,
        String email,
        String deviceId,
        String fcmToken,
        String deviceName,
        String platform
    ) {}

    record CreateUserDeviceTokenReq(
        String email,
        String deviceId,
        String fcmToken,
        String deviceName,
        String platform
    ) {}

    record UpdateUserDeviceTokenReq(
        String fcmToken,
        String deviceName,
        String platform
    ) {}

}

File: src/main/kotlin/com/lcaohoanq/sp/domains/notifications/FcmTokenPort.kt
package com.lcaohoanq.sp.domains.notifications

interface FcmTokenPort {

    data class UserDeviceTokenDto(
        val id: Long?,
        val userId: Long?,
        val deviceId: String?,
        val fcmToken: String?,
        val deviceName: String?,
        val platform: String?
    )

    data class CreateUserDeviceTokenReq(
        val userId: Long,
        val deviceId: String,
        val fcmToken: String,
        val deviceName: String?,
        val platform: String?
    )

    data class UpdateUserDeviceTokenReq(
        val fcmToken: String?,
        val deviceName: String?,
        val platform: String?
    )
}
